### PR TITLE
Erictr/todosample

### PR DESCRIPTION
**Pull request by erictraut**
This is a major revamp of the TodoList sample app. It incorporates a bunch of techniques we've used internally. See README for a full list.

Please review to make sure I'm not releasing anything that shouldn't go external.

### CODE REVIEW

File: samples/TodoList/src/ts/services/ServiceRegistrar.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+* ServiceRegistrar.tsx
```
Comment (deregtd) [inline samples/TodoList/src/ts/services/ServiceRegistrar.ts]: You moved this code here, but also have it in the base appbootstrapper

Comment (erictraut) [reply samples/TodoList/src/ts/services/ServiceRegistrar.ts]: Not sure what you mean. The ServiceRegistrar is registering all services. The AppBootstrapper is starting a few of them (well, actually all of them in the current sample).

**Comment by rolodex** [comment]
Has the Visual Studio sln been tested? It's failing due to SQLite plugin not being found.

**Comment by erictraut** [comment]
The react-native-sqlite-storage module needs to be updated to include Windows support. I used a non-published prerelease version to develop the sample. I'm waiting for @rigdern to add the code to the public repo and publish a new version. Once he does, the TodoList sample should work fine. Adam, do you have an update on this?

**Comment by rigdern** [comment]
I had to make some changes to the UWP react-native-sqlite-storage code while preparing it for the public repo. My goal is to send a react-native-sqlite-storage PR within the next couple of days.

**Pull request merged**